astropy/table/column.py

```diff
@@ -1393,6 +1395,10 @@ def __new__(cls, data=None, name=None, mask=None, fill_value=None,
                                unit=unit, format=format, description=description,
                                meta=meta, copy=copy, copy_indices=copy_indices)
         self = ma.MaskedArray.__new__(cls, data=self_data, mask=mask)
+        # The above process preserves info relevant for Column, but this does not
+        # include serialize_method relevant for MaskedColumn.
+        if isinstance(data, MaskedColumn) and 'info' in getattr(data, '__dict__', {}):
```

Comment:
I haven't thought about the details but why is `serialize_method` being singled out here instead of doing basically `self.info = data.info` to get any attributes that are there now or might be defined in the future? And why doesn't this get done already from the `Column` array finalize?

Sorry for being lazy and not reminding myself how this all works in detail...

Reply:
Mostly because the basecolumn did some of the `info` copying. But, really, you are right! I'll change that and check that it works.

Reply:
Oops, no, for this to work, we need to ensure info attribute names are not assumed to be identical, i.e., we need https://github.com/astropy/astropy/pull/11910/files#diff-72a70f056e1f7a7bad3e2e054326c7b17d2a815d0bf7a31b2bd59df2af039aa3R360

I've pulled it out of #11910 since it helps this PR we are now discussing (and is probably more obviously correct here). I'll rebase #11910 later.